src/cpp/session/modules/build/SessionBuildErrors.cpp

```diff
@@ -256,6 +256,92 @@ std::vector<module_context::SourceMarker> parseGccErrors(
    return errors;
 }
 
+std::vector<module_context::SourceMarker> parseTestThatErrors(
+                                           const FilePath& basePath,
+                                           const std::string& output)
+{
+   using namespace module_context;
+   std::vector<SourceMarker> errors;
+
+   try
+   {
+      FilePath basePathResolved = module_context::resolveAliasedPath(basePath.absolutePath());
+
+      boost::regex re("\\[1m([^:\\n]+):([0-9]+):[^:\\n]+: ([^\\n]*)\\[22m");
+
+      boost::sregex_iterator iter(output.begin(), output.end(), re);
+      boost::sregex_iterator end;
+      for (; iter != end; iter++)
+      {
+         boost::smatch match = *iter;
+         BOOST_ASSERT(match.size() == 4);
+
+         std::string file, line, type, message;
+         
+         file = match[1];
+         line = match[2];
+         type = "failure";
+         message = match[3];
+         FilePath testFilePath = basePathResolved.complete(file);
+
+         std::string column = "0";
+         SourceMarker err(module_context::sourceMarkerTypeFromString(type),
+                          testFilePath,
+                          core::safe_convert::stringTo<int>(line, 1),
+                          core::safe_convert::stringTo<int>(column, 1),
+                          core::html_utils::HTML(message),
+                          true);
+         errors.push_back(err);
+      }
+   }
+   CATCH_UNEXPECTED_EXCEPTION;
+
+   return errors;
+}
+
+std::vector<module_context::SourceMarker> parseShinyTestErrors(
+                                           const FilePath& basePath,
+                                           const std::string& output)
+{
+   using namespace module_context;
+   std::vector<SourceMarker> errors;
+
+   try
+   {
+      FilePath basePathResolved = module_context::resolveAliasedPath(basePath.absolutePath());
+
+      boost::regex re("Differences detected between ([^\\n]+)-current[/\\\\ ][^:]+");
```

Comment:
It feels like this could change easily in the package (it's not obvious that the IDE depends on this exact wording/spacing) -- maybe we should have the package give us control over the format of errors, and/or expose its error formatter?

Reply:
We can save the results as an `RDS` and then use that instead of parsing output, updated.

Reply:
Very cool!